src/main/scala/io/iohk/ethereum/consensus/blocks/BlockGeneratorImpl.scala

```diff
@@ -0,0 +1,174 @@
+package io.iohk.ethereum.consensus.blocks
+
+import java.util.concurrent.atomic.AtomicReference
+
+import akka.util.ByteString
+import io.iohk.ethereum.consensus.ConsensusConfig
+import io.iohk.ethereum.consensus.ethash.blocks.Ommers
+import io.iohk.ethereum.consensus.validators.std.MptListValidator.intByteArraySerializable
+import io.iohk.ethereum.crypto.kec256
+import io.iohk.ethereum.db.dataSource.EphemDataSource
+import io.iohk.ethereum.db.storage.{ArchiveNodeStorage, NodeStorage}
+import io.iohk.ethereum.domain._
+import io.iohk.ethereum.ledger.Ledger.{BlockPreparationResult, BlockResult}
+import io.iohk.ethereum.ledger.{BlockPreparator, BloomFilter}
+import io.iohk.ethereum.mpt.{ByteArraySerializable, MerklePatriciaTrie}
+import io.iohk.ethereum.network.p2p.messages.PV62.BlockBody
+import io.iohk.ethereum.network.p2p.messages.PV62.BlockHeaderImplicits._
+import io.iohk.ethereum.utils.BlockchainConfig
+import io.iohk.ethereum.utils.ByteUtils.or
+
+import scala.collection.immutable
```

Comment:
Is this on purpose? It's not used consistently here.

Reply:
The original

```scala
  private def prepareTransactions(transactions: Seq[SignedTransaction], blockGasLimit: BigInt) = { ...}
```
lacked a return type and when I had the IDE fill it in it was `List[SignedTransaction]`, so I guess that was how `immutable.Seq` appeared in the scene: Not too specific (a `List`) but definetly reflecting that fact that we are dealing with an `immutable.Seq`.